examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt

```diff
@@ -0,0 +1,237 @@
+package com.mapbox.navigation.examples.core
+
+import android.annotation.SuppressLint
+import android.location.Location
+import android.os.Bundle
+import android.preference.PreferenceManager
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.location.ReplayRouteLocationEngine
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import com.mapbox.navigation.ui.map.NavigationMapboxMapInstanceState
+import kotlinx.android.synthetic.main.activity_trip_service.mapView
+import kotlinx.android.synthetic.main.free_drive_navigation_layout.*
+import timber.log.Timber
+
+/**
+ * To ensure proper functioning of this example make sure your Location is turned on.
+ */
+class FreeDriveNavigationActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    companion object {
+        const val MAP_INSTANCE_STATE_KEY = "navgation_mapbox_map_instance_state"
+        const val DEFAULT_INTERVAL_IN_MILLISECONDS = 1000L
+        const val DEFAULT_MAX_WAIT_TIME = DEFAULT_INTERVAL_IN_MILLISECONDS * 5
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private var mapInstanceState: NavigationMapboxMapInstanceState? = null
+
+    @SuppressLint("MissingPermission")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.free_drive_navigation_layout)
+        mapView.onCreate(savedInstanceState)
+        mapView.getMapAsync(this)
+
+        val mapboxNavigationOptions = MapboxNavigation.defaultNavigationOptions(
```

Comment:
We should also configure the onboard router to actually get the benefits of the free drive.

Reply:
The MapboxNavigation.defaultNavigationOptions includes an onboard router in the configuration. Is there more that should be done?

Reply:
I added a tilesURI from the driver app.